### ISSUE TITLE
Overriding a default package version is not honoured for dependant packages (eg.  DEFAULT_VERSIONS=apache=2.2 does not build ap22-mod_php package, but ap24-mod_php instead)

### ISSUE THREAD

**Issue by ArminGruner** [jrmarino/synth#61]
I'm trying to maintain a local repository using synth under FreeBSD, both for setting some port options to a non-default value, and the package version to a non-default.
In /etc/make.conf, I've listed 

DEFAULT_VERSIONS+=apache=2.2

Though, when building for example www/mod_auth_external2, synth always will build 

ap24-mod_auth_external2.

If I manually do a "make" in  /usr/ports/www/mod_auth_external2 however, the proper apache version is used.

Same happens with www/mod_wsgi4, www/mod_php56.

If I manually change the default version in /usr/ports/mk/bsd.default-version.mk, the proper packages are built (using synth).

Any idea what's going on?

**Comment by jrmarino** [comment]
yes.  refer to the man page "man 1 synth", FILES section, <profile>-make.conf

tldr: synth doesn't look at /etc/make.conf.  It's like it doesn't exist.

**Comment by jrmarino** [comment]
also look here: https://github.com/jrmarino/synth#custom-makeconf

**Comment by ArminGruner** [comment]
Okay, I must have been msireading this section:

<profile>-make.conf
                       This is an optional, user-provided file. If it exists,
                       the builder's /etc/make.conf will be appended with the
                       contents of this file. For the default profile, the
                       file would normally be located at
                       /usr/local/etc/synth/LiveSystem-make.conf

Now I've linked /usr/local/etc/synth/LiveSystem-make.conf  -> /etc/make.conf and I get what is expected.  (though, I still have some issues with local ports, I will raise another issue separately)

Thanks for the quick feedback and Greets from Munich!

...

**Comment by jrmarino** [comment]
Please don't do that, it defeats the purpose and a really bad idea.

if you have to, copy /etc/make.conf over and remove the cruft.

I miss living in Munich. :(

**Comment by jrmarino** [comment]
another technique is putting this in /etc/make.conf:

```
.include "/usr/local/etc/synth/LiveSystem-make.conf
```

and move the ports stuff from /etc/make.conf to LiveSystem-make.conf

**Comment by ArminGruner** [comment]
Hm, I must oversee something. Actually the "livesystem" profile *IS* the setting for my "live system", so I actually *want* all my /etc/make.conf settings applied when synth builds a repository which also reflects all my system base ports with its options?

**Comment by jrmarino** [comment]
it's not.  /etc/make.conf also applies to the actual system, not just ports. 

if you absolutely have to have ports settings in one place, use the suggestion with ".include" above.

symlinks defeats the purpose of each profile having a pristine make.conf.

I can't stop you from symlinking, I can only heavy recommend against it by calling it a really bad idea (TM).
